deeptime/markov/msm/tram/_tram_dataset.py

```diff
@@ -19,6 +19,18 @@ def _determine_n_therm_states(dtrajs, ttrajs):
         return _determine_n_states(ttrajs)
 
 
+def _split_at_negative_state_indices(trajectory_fragment, negative_state_indices):
+    split_fragments = np.split(trajectory_fragment, negative_state_indices)
+    sub_fragments = []
+    # now get rid of the negative state indices.
+    for frag in split_fragments:
+        frag = frag[frag >= 0]
+        # Only add to the list if there are any samples left in the fragments
+        if len(frag) > 0:
```

Comment:
You can check for `>= lagtime` here

Reply:
No, because the state counts are still counted  even if there are no countable transitions due to the lag time

Reply:
Right makes sense, merging